plugins/webp-uploads/tests/test-picture-element.php

```diff
@@ -11,40 +11,26 @@
 
 class Test_WebP_Uploads_Picture_Element extends TestCase {
 
-	/**
-	 * Attachment ID.
-	 *
-	 * @var int
-	 */
-	public static $image_id;
-
-	/**
-	 * Setup shared fixtures.
-	 */
-	public static function wpSetUpBeforeClass( WP_UnitTest_Factory $factory ): void {
```

Comment:
@joemcgill Initially i implemented it but if we use the shared attachment then it will return WEBP image instead of JPEG.

```
<img width="1024" height="683" src="http://localhost:8889/wp-content/uploads/2024/08/leaves-417-1024x683.webp" class="wp-image-4" alt="Green Leaves" decoding="async" loading="lazy" srcset="http://localhost:8889/wp-content/uploads/2024/08/leaves-417-1024x683.webp 1024w, http://localhost:8889/wp-content/uploads/2024/08/leaves-417-300x200.webp 300w, http://localhost:8889/wp-content/uploads/2024/08/leaves-417-768x512.webp 768w, http://localhost:8889/wp-content/uploads/2024/08/leaves-417-jpg.webp 1080w" sizes="(max-width: 1024px) 100vw, 1024px" />
```

Per https://web.dev/learn/design/picture-element we should fallback to JPEG so i use new attachment for each image.

```
<picture>
  <source srcset="image.avif" type="image/avif">
  <source srcset="image.webp" type="image/webp">
  <img src="image.jpg" alt="A description of the image." 
    width="300" height="200" loading="lazy" decoding="async">
</picture>
```

Reply:
Got it. We need to set the fallback option before creating the shared attachment. See https://github.com/WordPress/performance/pull/1408#discussion_r1704383543.